Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -738,27 +738,35 @@ internal extension ByteBuffer {
         self.write(staticString: crlf)
     }
 }
-extension HTTPHeaders: Sequence {
-    public typealias Element = (name: String, value: String)
-
-    /// An iterator of HTTP header fields.
-    ///
-    /// This iterator will return each value for a given header name separately. That
-    /// means that `name` is not guaranteed to be unique in a given block of headers.
-    public struct Iterator: IteratorProtocol {
-        private var headerParts: Array<(String, String)>.Iterator
-
-        fileprivate init(headerParts: Array<(String, String)>.Iterator) {
-            self.headerParts = headerParts
-        }
 
-        public mutating func next() -> Element? {
-            return headerParts.next()
+extension HTTPHeaders: RandomAccessCollection {
+    public typealias Element = (name: String, value: String)
+    public struct Index: Comparable {
+        let base: Int
```

Comment:
Let's make this `fileprivate let` to avoid the rest of the project being able to see it.